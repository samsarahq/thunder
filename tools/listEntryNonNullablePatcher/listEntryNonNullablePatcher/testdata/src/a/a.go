package a

import (
	"github.com/samsarahq/thunder/batch"
)

type Server struct{}

type objectType struct{}

func schema() *Schema {
	server := &Server{}
	schema := NewSchema()
	server.Register(schema)
	return schema
}

func (s *Server) Register(schema *Schema) {
	s.registerRoot(schema)
}

func (s *Server) registerRoot(schema *Schema) {
	object := schema.Query()
	object.FieldFunc("listEntryPtr", s.listEntryPtr)
	object.FieldFunc("listEntryPtrDone", s.listEntryPtr)
	object.FieldFunc(
		"listEntryPtr",
		s.listEntryPtr,
	)
	object.FieldFunc("listEntryNonPtr", s.listEntryNonPtr)
	object.FieldFunc("listEntryCustomTypePtr", s.listEntryCustomTypePtr)
	object.FieldFunc("listEntryCustomSliceTypePtr", s.listEntryCustomSliceTypePtr)
	object.FieldFunc("listEntryPtrAnonymous", func() []*string { return []*string{} })
	object.FieldFunc("listEntryNonPtrAnonymous", func() []string { return []string{} })
	object.FieldFunc("listEntryPtrNormal", listEntryPtr)
	object.FieldFunc("listEntryNonPtrNormal", listEntryNonPtr)
	object.FieldFunc("listEntryCustomTypePtrNormal", listEntryCustomTypePtr)
	object.FieldFunc("listEntryCustomSliceTypePtrNormal", listEntryCustomSliceTypePtr)

	object.FieldFunc("batchListEntryPtr", s.batchListEntryPtr)
	object.FieldFunc("batchListEntryNonPtr", s.batchListEntryNonPtr)
	object.FieldFunc("batchListEntryPtrNormal", batchListEntryPtr)
	object.FieldFunc("batchListEntryNonPtrNormal", batchListEntryNonPtr)
	object.FieldFunc("batchListEntryPtrAnonymous", func() map[batch.Index][]*string { return map[batch.Index][]*string{} })
	object.FieldFunc("batchListEntryNonPtrAnonymous", func() map[batch.Index][]string { return map[batch.Index][]string{} })
	object.FieldFunc("batchListEntryCustomSliceTypePtr", s.batchListEntryCustomSliceTypePtr)
	object.FieldFunc("batchListEntryCustomSliceTypePtrNormal", batchListEntryCustomSliceTypePtr)
}

type myStringPtr *string

func (s *Server) listEntryCustomTypePtr() []myStringPtr {
	return []myStringPtr{}
}

type myStringPtrSlice []*string

func (s *Server) listEntryCustomSliceTypePtr() myStringPtrSlice {
	return myStringPtrSlice{}
}

type myMapPtr map[batch.Index]myStringPtrSlice

func (s *Server) batchListEntryCustomSliceTypePtr() myMapPtr {
	return myMapPtr{}
}

func (s *Server) listEntryPtr() []*string {
	return []*string{}
}

func (s *Server) listEntryNonPtr() []string {
	return []string{}
}

func (s *Server) batchListEntryPtr() map[batch.Index][]*string {
	return map[batch.Index][]*string{}
}

func (s *Server) batchListEntryNonPtr() map[batch.Index][]string {
	return map[batch.Index][]string{}
}

func listEntryPtr() []*string {
	return []*string{}
}

func listEntryNonPtr() []string {
	return []string{}
}

func batchListEntryPtr() map[batch.Index][]*string {
	return map[batch.Index][]*string{}
}

func batchListEntryNonPtr() map[batch.Index][]string {
	return map[batch.Index][]string{}
}

func listEntryCustomTypePtr() []myStringPtr {
	return []myStringPtr{}
}

func listEntryCustomSliceTypePtr() myStringPtrSlice {
	return myStringPtrSlice{}
}

func batchListEntryCustomSliceTypePtr() myMapPtr {
	return myMapPtr{}
}
