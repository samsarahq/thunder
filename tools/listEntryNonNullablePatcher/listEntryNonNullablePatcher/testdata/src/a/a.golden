package a

import (
	"github.com/samsarahq/thunder/batch"
)

type Server struct{}

type objectType struct{}

func schema() *Schema {
	server := &Server{}
	schema := NewSchema()
	server.Register(schema)
	return schema
}

func (s *Server) Register(schema *Schema) {
	s.registerRoot(schema)
}

func (s *Server) registerRoot(schema *Schema) {
	object := schema.Query()
	object.FieldFunc("listEntryPtr", s.listEntryPtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("listEntryPtrDone", s.listEntryPtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc(
		"listEntryPtr",
		s.listEntryPtr,
		schemabuilder.ListEntryNonNullable)
	object.FieldFunc("listEntryNonPtr", s.listEntryNonPtr)
	object.FieldFunc("listEntryCustomTypePtr", s.listEntryCustomTypePtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("listEntryCustomSliceTypePtr", s.listEntryCustomSliceTypePtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("listEntryPtrAnonymous", func() []*string { return []*string{} }, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("listEntryNonPtrAnonymous", func() []string { return []string{} })
	object.FieldFunc("listEntryPtrNormal", listEntryPtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("listEntryNonPtrNormal", listEntryNonPtr)
	object.FieldFunc("listEntryCustomTypePtrNormal", listEntryCustomTypePtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("listEntryCustomSliceTypePtrNormal", listEntryCustomSliceTypePtr, schemabuilder.ListEntryNonNullable)

	object.FieldFunc("batchListEntryPtr", s.batchListEntryPtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("batchListEntryNonPtr", s.batchListEntryNonPtr)
	object.FieldFunc("batchListEntryPtrNormal", batchListEntryPtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("batchListEntryNonPtrNormal", batchListEntryNonPtr)
	object.FieldFunc("batchListEntryPtrAnonymous", func() map[batch.Index][]*string { return map[batch.Index][]*string{} }, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("batchListEntryNonPtrAnonymous", func() map[batch.Index][]string { return map[batch.Index][]string{} })
	object.FieldFunc("batchListEntryCustomSliceTypePtr", s.batchListEntryCustomSliceTypePtr, schemabuilder.ListEntryNonNullable)
	object.FieldFunc("batchListEntryCustomSliceTypePtrNormal", batchListEntryCustomSliceTypePtr, schemabuilder.ListEntryNonNullable)
}

type myStringPtr *string

func (s *Server) listEntryCustomTypePtr() []myStringPtr {
	return []myStringPtr{}
}

type myStringPtrSlice []*string

func (s *Server) listEntryCustomSliceTypePtr() myStringPtrSlice {
	return myStringPtrSlice{}
}

type myMapPtr map[batch.Index]myStringPtrSlice

func (s *Server) batchListEntryCustomSliceTypePtr() myMapPtr {
	return myMapPtr{}
}

func (s *Server) listEntryPtr() []*string {
	return []*string{}
}

func (s *Server) listEntryNonPtr() []string {
	return []string{}
}

func (s *Server) batchListEntryPtr() map[batch.Index][]*string {
	return map[batch.Index][]*string{}
}

func (s *Server) batchListEntryNonPtr() map[batch.Index][]string {
	return map[batch.Index][]string{}
}

func listEntryPtr() []*string {
	return []*string{}
}

func listEntryNonPtr() []string {
	return []string{}
}

func batchListEntryPtr() map[batch.Index][]*string {
	return map[batch.Index][]*string{}
}

func batchListEntryNonPtr() map[batch.Index][]string {
	return map[batch.Index][]string{}
}

func listEntryCustomTypePtr() []myStringPtr {
	return []myStringPtr{}
}

func listEntryCustomSliceTypePtr() myStringPtrSlice {
	return myStringPtrSlice{}
}

func batchListEntryCustomSliceTypePtr() myMapPtr {
	return myMapPtr{}
}
